src/Microsoft.Extensions.ClosedGenericMatcher.Sources/ClosedGenericMatcher.cs

```diff
@@ -40,16 +40,63 @@ public static Type ExtractGenericInterface(Type queryType, Type interfaceType)
                 throw new ArgumentNullException(nameof(interfaceType));
             }
 
-            Func<Type, bool> matchesInterface =
-                type => type.GetTypeInfo().IsGenericType && type.GetGenericTypeDefinition() == interfaceType;
-            if (matchesInterface(queryType))
+            // Checked type matches (i.e. is a closed generic type created from) the open generic type.
+            if (IsGenericInstantiation(queryType, interfaceType))
```

Comment:
Will move comment back inside the `if`. I refactored this a few times, and didn't realise I had moved it.